docs/examples/wikitext.ts

```diff
@@ -17,56 +15,54 @@ async function main(): Promise<void> {
-      fs.mkdirSync(modelFolder)
-    }
-  } catch (err) {
-    console.error(err);
-  }
-  const encoded = await serialization.model.encode(model)
-  await fsPromises.writeFile(`${modelFolder}/model.json`, encoded)
+  const modelFileName = 'model_random.json'
   
-  /**
-   * The commented code below shows how to load an existing model
-   */
-
-  // const content = await fsPromises.readFile(`${modelFolder}/model.json`)
-  // model = await serialization.model.decode(content) as models.GPT
-  // console.log(await model.generate(sample, tokenizerName, 10))
+  // Toggle TRAIN_MODEL to either train and save a new model from scratch or load an existing model
+  const TRAIN_MODEL = true
+  if (TRAIN_MODEL) {
+    // Load the wikitext dataset from the `datasets` folder
+    const dataset = await loadWikitextData(task)
+  
+    // Initialize a Disco instance and start training a language model
+    const aggregator = new aggregators.MeanAggregator()
+    const client = new clients.federated.FederatedClient(url, task, aggregator)
+    const trainingInformant = new informant.FederatedInformant(task, 10)
+    const disco = new Disco(task, { scheme: 'federated', client, aggregator, informant: trainingInformant })
+    await disco.fit(dataset)
+  
+    // Get the model and complete the prompt
+    if (aggregator.model === undefined) {
+      throw new Error('model was not set')
+    }
+    // Save the trained model
+    const model = aggregator.model as models.GPT
+    await saveModelToDisk(model, modelFolder, modelFileName)
+    
+    // Retrieve the tokenizer used during training
+    const tokenizer = await models.getTaskTokenizer(task)
+    const prompt = 'The game began development in 2010 , carrying over a large portion'
+    const generations = await model.generate(prompt, tokenizer)
+    console.log(generations)
+    
+    await disco.close()
+  } else {
+    // Load the trained model
+    const model = await loadModelFromDisk(`${modelFolder}/${modelFileName}`) as models.GPT
 
-  await disco.close()
+    // Retrieve the tokenizer used during training
+    const tokenizer = await models.getTaskTokenizer(task)
+    // The game began development in 2010 , carrying over a large portion
+    const prompt = 'Hello world how'
+    console.log(await model.generate(prompt, tokenizer, 20))
+  }
 }
 
 async function loadWikitextData (task: Task): Promise<data.DataSplit> {
   const loader = new NodeTextLoader(task)
   const dataSplit: data.DataSplit = {
-    train: await data.TextData.init(await loader.load('../../datasets/wikitext/wiki.train.tokens', {shuffle: false}), task),
-    validation: await data.TextData.init(await loader.load('../../datasets/wikitext/wiki.valid.tokens', {shuffle: false}), task)
+    train: await data.TextData.init(await loader.load('../../datasets/wikitext/wiki.train.tokens', {shuffle: true}), task),
+    validation: await data.TextData.init(await loader.load('../../datasets/wikitext/wiki.valid.tokens', {shuffle: true}), task)
```

Comment:
it's shuffling by default
```suggestion
    train: await data.TextData.init(await loader.load('../../datasets/wikitext/wiki.train.tokens'), task),
    validation: await data.TextData.init(await loader.load('../../datasets/wikitext/wiki.valid.tokens'), task)
```

Reply:
I want to be make it explicit in the example, I found out myself way too late that shuffling was happening behind the scene